### PR TITLE
Fix: Prepare config before verifying SourceCode (fixes #4230)

### LINKED ISSUE #4230: `eslint.verify` does not prepare config if SourceCode is used
When a `SourceCode` object is passed into `eslint.verify`, `prepareConfig` is not executed, causing an error like:

```
/home/ian/Local_code/eslint/eslint/lib/eslint.js:721
                    throw ex;
                    ^

TypeError: Error while loading rule 'accessor-pairs': Object prototype may only be an Object or null: undefined
    at Function.create (native)
    at new RuleContext (/home/ian/Local_code/eslint/eslint/lib/rule-context.js:101:23)
    at /home/ian/Local_code/eslint/eslint/lib/eslint.js:707:40
    at Array.forEach (native)
    at EventEmitter.module.exports.api.verify (/home/ian/Local_code/eslint/eslint/lib/eslint.js:686:16)
```

I believe the lines https://github.com/eslint/eslint/blob/master/lib/eslint.js#L660-L661 should be moved outside of the if block they are currently in.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27629866-eslint-verify-does-not-prepare-config-if-sourcecode-is-used?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by IanVS**

### CODE REVIEW

File: tests/lib/eslint.js [hunk 1]
```diff
@@ -959,6 +959,21 @@ describe("eslint", function() {
...
+
+        it("SourceCode objects should be processed", function() {
+            var config = {
+                rules: {},
+                ecmaFeatures: {}
+            };
+
+            eslint.reset();
+            eslint.verify(code, config, filename, true);
+            var sourceCode = eslint.getSourceCode();
```
Comment (nzakas) [inline tests/lib/eslint.js]: This is a really roundabout way to get a `SourceCode` object, so it's not very clear what this test is doing. Can you just manually create it so you're not calling `verify()` twice?

File: tests/lib/eslint.js [hunk 2]
```diff
@@ -1088,6 +1103,18 @@ describe("eslint", function() {
             var messages = eslint.verify(code, config, filename, true);
             assert.equal(messages.length, 0);
         });
+
+        it("should process SourceCode with empty config", function() {
+            var config = {};
+
+            eslint.reset();
+            eslint.verify(code, config, filename, true);
+            var sourceCode = eslint.getSourceCode();
```
Comment (nzakas) [inline tests/lib/eslint.js]: Ditto

File: tests/lib/eslint.js (hunk 1 above)
Comment (IanVS) [inline tests/lib/eslint.js]: Yes, I didn't think about `eslint.parse()`, which I see now you are using in the test for `source-code.js`.  That's a much better way.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**